src/iproject.ts

```diff
@@ -68,39 +106,127 @@ export class IProject {
     return Uri.file(path.join(this.workspaceFolder.uri.fsPath, `.env`));
   }
 
-  public async read() {
-    const content = await workspace.fs.readFile(this.getIProjFilePath());
-    this.state = IProject.validateIProject(content.toString());
+  public async addToIncludePaths(includePath: string) {
+    const unresolvedState = await this.getUnresolvedState();
+
+    if (unresolvedState) {
+      if (unresolvedState.includePath) {
+        if (!unresolvedState.includePath.includes(includePath)) {
+          unresolvedState.includePath.push(includePath);
+        } else {
+          window.showErrorMessage(`${includePath} already exists in includePaths`);
+          return;
+        }
+      } else {
+        unresolvedState.includePath = [includePath];
+      }
+
+      await this.updateIProj(unresolvedState);
+    } else {
+      window.showErrorMessage('No iproj.json found');
+    }
   }
 
-  public async addToIncludePaths(includePath: string) {
-    const iProjExists = await this.projectFileExists('iproj.json');
-    if (iProjExists) {
-      const content = await workspace.fs.readFile(this.getIProjFilePath());
+  public async addToLibraryList(library: string, position: 'preUsrlibl' | 'postUsrlibl') {
+    const unresolvedState = await this.getUnresolvedState();
+    const state = await this.getState();
+
+    if (unresolvedState && state) {
+      if (unresolvedState[position] && state[position]) {
+        if (state[position]!.includes(library)) {
+          window.showErrorMessage(`${library} already exists in ${position}`);
+          return;
 
-      const iProject = IProject.validateIProject(content.toString());
-      if (iProject) {
-        try {
-          if (iProject.includePath) {
-            if (!iProject.includePath.includes(includePath)) {
-              iProject.includePath.push(includePath);
-            } else {
-              window.showErrorMessage(`${includePath} already exists in includePaths`);
+        } else {
+          unresolvedState[position]!.push(library);
```

Comment:
do we insert additions to the preUsrLibl at the beginning of the list and 
append additions to the postUsrLibl at the end of the list

Reply:
Good catch. I pushed a fix for this